src/backports/three_way_comparison.h

```diff
@@ -0,0 +1,384 @@
+//
+// Created by kalmbacj on 9/2/25.
+//
+// Copyright 2025, Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+
+#ifndef THREE_WAY_COMPARISON_H
+#define THREE_WAY_COMPARISON_H
+
+#include <cmath>
+#include <tuple>
+#include <type_traits>
+
+#ifdef QLEVER_CPP_17
+
+namespace ql {
+
+using strong_ordering = std::strong_ordering;
+using partial_ordering = std::partial_ordering;
+using weak_ordering = std::weak_ordering;
```

Comment:
There is `absl::strong_ordering` etc, which is meant as a drop-in replacement for `std::strong_ordering` etc, you can use it here.